functionary/core/utils/parameters.py

```diff
@@ -0,0 +1,37 @@
+from collections import defaultdict, namedtuple
+
+Parameter = namedtuple("Parameter", "type format", defaults=(None, None))
+
+
+def parameter_mapping(schema: dict) -> dict[str, list[Parameter]]:
```

Comment:
It's worth pointing out that the parameter handling changes I'm making to get workflow and function parameters working the same is going to make this either change completely or go away entirely.  If other schema handling stuff comes up as part of this effort, don't spend any time trying to make it something wonderful, since it's all going to go away very soon.